src/app/subject/services/volume-data.service.ts

```diff
@@ -0,0 +1,74 @@
+import { HttpClient, HttpHeaders } from '@angular/common/http'
+import { Injectable } from '@angular/core'
+import { Actions, ofType } from '@ngrx/effects'
+import { Action } from '@ngrx/store'
+import { Observable } from 'rxjs'
+import { takeUntil } from 'rxjs/operators'
+
+import { ENV } from '../../../environments/environment'
+import { DescriptiveStatistic } from '../../shared/enums/descriptive-statistic.enum'
+import { Source } from '../../shared/models/source.model'
+import { Subject } from '../../shared/models/subject.model'
+import * as volumeDataActions from '../store/volume-data/volume-data.actions'
+
+@Injectable()
+export class VolumeDataService {
+  private url = `${ENV.API_URI}/aggregate`
+  private destroy$: Observable<Action>
+
+  // NOTE: Temporary values are used to get data from volume API. Time window is automatically specified (ONE_WEEK) by API.
+  private endTime = '2018-04-27T14:00:00Z'
+  private startTime = '2017-04-27T12:00:00Z'
+
+  constructor(private http: HttpClient, private actions$: Actions) {
+    this.destroy$ = this.actions$.pipe(ofType(volumeDataActions.DESTROY))
+  }
+
+  getData(
+    studyName,
+    subjectId,
+    descriptiveStatistic,
+    sources: any
+  ): Observable<any> {
+    const new_sources = new Array()
+
+    for (const key of Object.keys(sources)) {
+      const new_sourcedata = new Array()
+      for (const data of Object.keys(sources[key].sourceData)) {
+        new_sourcedata.push({
+          name: sources[key].sourceData[data].sourceDataName
+        })
+      }
+      new_sources.push({
+        sourceId: sources[key].sourceId,
+        sourceData: new_sourcedata
+      })
+    }
+
+    return this.http
+      .post<any>(
+        this.parseURL(studyName, subjectId, descriptiveStatistic),
+        JSON.stringify({ sources: new_sources }),
+        {
+          headers: new HttpHeaders({
+            'Content-Type': 'application/json'
+          })
+        }
+      )
+      .pipe(takeUntil(this.destroy$))
+  }
+
+  private parseURL(studyName, subjectId, descriptiveStatistic): string {
+    let url = [
+      this.url,
+      studyName,
+      subjectId,
+      DescriptiveStatistic[descriptiveStatistic].toLowerCase()
+    ].join('/')
+
+    this.startTime ? (url = `${url}?startTime=${this.startTime}`) : (url = url)
```

Comment:
Added custom startTime and endTime because without it, I'm still getting 500 Internal Server Error

Reply:
@yatharthranjan do you know which version or rest-api we are using at rosalind?

Reply:
it was pulled from dev two week ago. So i am guessing just after the sorting fix. Do you want me to update it to latest dev or should i wait till a release ?

Reply:
yes. Please. Or we could base the development of dashboard on test env